.github/workflows/examples.yml

```diff
@@ -35,34 +34,30 @@ jobs:
           mkdir -p ~/.terraform.d/plugins/registry.terraform.io/hashicorp/mongodbatlas-dev/99.99.99/"$(go env GOOS)"_"$(go env GOARCH)"/
           cp terraform-plugin-dir/terraform-provider-mongodbatlas-dev_v99.99.99_x5 ~/.terraform.d/plugins/registry.terraform.io/hashicorp/mongodbatlas-dev/99.99.99/"$(go env GOOS)"_"$(go env GOARCH)"/
       - name: replace names of provider for local development
-        run: grep --include=\*.tf -rnl './examples' -e 'source = "mongodb/mongodbatlas"'  | xargs  sed -i s@mongodb/mongodbatlas@hashicorp/mongodbatlas-dev@g
+        run: grep --include=\*.tf -rnl './examples' -e 'source  = "mongodb/mongodbatlas"' | grep -vE "test-upgrade" | xargs sed -i s@mongodb/mongodbatlas@hashicorp/mongodbatlas-dev@g
+      - name: remove version of provider for local development
+        run: grep --include=\*.tf -rnl './examples' -e 'version =' | grep -vE "test-upgrade" | xargs sed -i '/^\s*version =/d'
+      - name: Fix format after sostitutions
+        run: grep --include=\*.tf -rnl './examples' -e 'source[[:space:]]\+=' | grep -vE "test-upgrade" | xargs sed -i 's@\(\([[:space:]]*\)source\)[[:space:]]\+=[[:space:]]*@\2source = @g'
       - uses: hashicorp/setup-terraform@v2
         with:
           terraform_version: ${{ matrix.terraform_version }}
           # Needed to use the output of `terraform validate -json`
           terraform_wrapper: false
-      - name: install tflint
-        run: go install github.com/terraform-linters/tflint@v0.31.0
-      - name: terraform
-        run: |
-          for DIR in $(find ./examples -type f -name '*.tf' -exec dirname {} \; | sort -u); do
-              pushd "${DIR}"
-              if [ -f terraform.template.tfvars ]; then
-                cp terraform.template.tfvars terraform.tfvars
-              fi
-              echo; echo -e "\e[1;35m===> Initializing Example: $DIR <===\e[0m"; echo
-              terraform init
-              echo; echo -e "\e[1;35m===> Format Checking Example: $DIR <===\e[0m"; echo
-              terraform fmt -check
-              echo; echo -e "\e[1;35m===> Validating Example: $DIR <===\e[0m"; echo
-              # Catch errors
-              terraform validate
-              # Terraform syntax checks
-              tflint \
-                --enable-rule=terraform_deprecated_interpolation \
-                --enable-rule=terraform_deprecated_index \
-                --enable-rule=terraform_unused_declarations \
-                --enable-rule=terraform_comment_syntax \
-                --enable-rule=terraform_required_version
-              popd
-            done
+      - name: tf-validate
```

Comment:
I have divided the original job into two separate jobs, each with a specific purpose:
- tf-validate: 
   - This job executes three Terraform commands: `terraform init`, `terraform fmt`, and `terraform validate`
    - It is necessary to replace the provider with `mongodbatlas-dev` in this job. This is because new resources are not available until the next release.
    - It is important to note that when using `mongodbatlas-dev` as the provider, the `version` attribute cannot be specified for the dev provider. However, `tflint` mandates that the `version` must be set for all providers in the Terraform configuration. This discrepancy is the reason for dividing the initial job.
 - tflint: this job runs `tflint`